src/content/tools/hooks.md

```diff
@@ -0,0 +1,277 @@
+at runtime from the Dart code. Currently, the Dart SDK can
+use the `CodeAsset` type, but more are planned.
+To learn more, see the following.
+
+### CodeAsset type {:.no_toc}
+
+A [`CodeAsset`][] represents a code asset. A code asset is
+a dynamic library compiled from a language other than
+Dart, such as C, C++, Rust, or Go. `CodeAsset` is part of
+the `code_asset` package and are called with `dart:ffi` from
+the Dart code at runtime using the `@Native` annotation.
+A code asset can be used with build and link hooks.
+
+[`CodeAsset`]: {{site.pub-api}}/code_assets/latest/code_assets/CodeAsset-class.html
+
+## Use a hook {: #use-hooks-assets }
+
+To add assets to your project, use a hook. For details,
+see the following sections.
+
+### Add dependencies {: #add-dependencies-hooks-assets }
+
+Open your package in a code editor, navigate to its
+`pubspec.yaml`, and then add the following dependencies as
+needed under `dependencies`:
+
+* `hooks`
+* `code_assets`
+
+You will also likely need to add a tool for your specific
+task, such as `native_toolchain_c` for compiling C code.
+
+:::note
+You need to add the dependencies under `dependencies`, not
+`dev_dependencies`. The hooks will be run by packages and
+applications depending on your package, so the
+Dart code needs to be part of the resolution in those
+packages.
+:::
+
+Example dependencies for a build hook:
+
+```yaml title="pubspec.yaml"
+name: native_add_library
+description: Sums two numbers with native code.
+version: 0.1.0
+
+environment:
+  sdk: '>=3.9.0 <4.0.0'
+
+dependencies:
+  ... 
+  code_assets: any
+  hooks: any
+  native_toolchain_c: any
+
+dev_dependencies:
+  ...
+  ffigen: ^18.0.0
+ ```
+
+### Create a build hook script {: #create-hook }
+
+If you want to use a build hook, first create the native
+library that you want to use, and then follow these steps to
+connect a build hook to that library in a Dart script: 
+
+1.  In your Dart project, create or open `hooks/build.dart`. 
+
+1.  In the main method, add a `build` function from
+    `package:hooks/hooks.dart` and link to the native
+    library. For example:
+
+    ```dart title="hooks/build.dart" highlightLines=6
+    import 'package:hooks/hooks.dart';
+    import 'package:logging/logging.dart';
+    import 'package:native_toolchain_c/native_toolchain_c.dart';
+
+    void main(List<String> args) async {
+      await build(args, (input, output) async {
+        final packageName = input.packageName;
+        final cbuilder = CBuilder.library(
+          name: packageName,
+          assetName: '$packageName.dart',
+          sources: ['src/$packageName.c'],
+        );
+        await cbuilder.run(
+          input: input,
+          output: output,
+          logger: Logger('')
+            ..level = Level.ALL
+            ..onRecord.listen((record) => print(record.message)),
+        );
+      });
+    }
+    ```
+
+    * `input`: The input for `hook/build.dart`. This
```

Comment:
Maybe it would be useful to include that the `input` is read-only and it includes information for the hook to produce the right asset type, e.g. like target OS etc.

Reply:
Yes, definitely. Updated!